src/Features/Core/Portable/EditAndContinue/CommittedSolution.cs

```diff
@@ -458,16 +455,22 @@ await TryGetMatchingSourceTextAsync(log, sourceText, sourceFilePath, currentDocu
         }
     }
 
-    public void CommitChanges(Solution solution, ImmutableArray<ProjectId> projectsToStale, IReadOnlyCollection<ProjectId> projectsToUnstale)
+    public void CommitChanges(Solution solution, ImmutableDictionary<ProjectId, Guid>? staleProjects, ImmutableArray<ProjectId>? projectsToUnstale = null)
     {
-        Debug.Assert(projectsToStale.Intersect(projectsToUnstale).IsEmpty());
+        Contract.ThrowIfTrue(staleProjects is null && projectsToUnstale is null);
```

Comment:
>Contract.ThrowIfTrue(staleProjects is null && projectsToUnstale is null); [](http://example.com/codeflow?start=8&length=73)

Would it make sense to validate one of the args is null?

Reply:
The second arg is temporary. Will be removed soon.